x/concentrated-liquidity/pool.go

```diff
@@ -215,9 +215,12 @@ func (k Keeper) CalcOutAmtGivenIn(ctx sdk.Context,
 			// update the swapState's liquidity with the new tick's liquidity
 			newLiquidity := addLiquidity(swapState.liquidity, liquidityDelta)
 			swapState.liquidity = newLiquidity
-			if swapState.liquidity.LTE(sdk.ZeroDec()) || swapState.liquidity.IsNil() {
-				return sdk.Coin{}, sdk.Coin{}, sdk.Int{}, sdk.Dec{}, sdk.Dec{}, err
-			}
+
+			// TODO: Investigate proper break of loop given the following conditions
```

Comment:
Do the new test cases fail when we enable these checks?

Reply:
Specifically the test case where there is a break in position. So if we have a 4545-5500 and a 5502-6750 position, if we have this check then we we stop at tick 5500 instead of looking for another tick (in this case 5502), which imo is a bug

Reply:
Agreed on context loss concern

This error path was never hit in all other test cases because all other test cases have overlapping ticks. When ticks overlap, liquidity can bleed into the next. The problem occurred with non overlapping fields where we use the entire liquidity of the upper tick, then we need to skip past empty ticks to the next initialized tick. I will need to think about this more to figure out what we should do

Reply:
As discussed offline, included link to this discussion in code comment